build/Dockerfile

```diff
@@ -1,22 +1,22 @@
 # Build stage
 FROM --platform=${BUILDPLATFORM:-linux/amd64} golang:1.15.4-alpine AS builder
-COPY ./.git promscale/.git
-COPY ./pkg promscale/pkg
-COPY ./cmd promscale/cmd
-COPY ./go.mod promscale/go.mod
-COPY ./go.sum promscale/go.sum
+RUN apk update && apk add --no-cache git
```

Comment:
Switching images from `golang:1.15.4-alpine` to `golang:1.15.4` would render `git` installation unnecessary and thus speed things even more. 

We don't need to use `-alpine` variant as this is just a builder image.

We should probably switch to even newer image for golang as we are using `1.16` in tests.

Reply:
Good point but I'd rather keep the scope of this PR limited. Happy to have another PR for changing base image (and also deciding on which one to pick :)